ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/MetaStoreCompactorThread.java

```diff
@@ -101,6 +103,25 @@ public void init(AtomicBoolean stop) throws Exception {
     }
   }
 
+  @Override
+  public void setCache(CompactorMetadataCache metadataCache) {
+    this.metadataCache = metadataCache;
+  }
+
+  protected Table cacheAndResolveTable(CompactionInfo ci) throws MetaException {
+    if (metadataCache != null) {
```

Comment:
do you think it's worth using Optional here?

Reply:
Well I was thinking this previously and because there was no default value for the cache, I decided to allow it as null if absent. But you are right in a sense, that using optional explicitly says that the cache can be null. 
Fixed.